docs/topics/compatibility-guides/compatibility-guide-22.md

```diff
@@ -0,0 +1,574 @@
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: deprecate the use of REPL in `kotlinc` with a warning
+> - 2.2.0: to use REPL via `kotlinc`, opt in with the `-Xrepl` compiler option; deprecate JSR-223, support can be restored by switching to language version 1.9; deprecate the `KotlinScriptMojo` Maven plugin
+
+### Deprecate disambiguation classifier properties
+
+> **Issue**: [KT-58231](https://youtrack.jetbrains.com/issue/KT-58231)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Options that were used to control how the Kotlin Gradle plugin disambiguates source set names and
+> IDE imports became obsolete. Therefore, the following properties are now deprecated in the `KotlinTarget` interface:
+>
+> * `useDisambiguationClassifierAsSourceSetNamePrefix`
+> * `overrideDisambiguationClassifierOnIdeImport`
+>
+> **Deprecation cycle**:
+>
+> - 2.0.0: report a warning when the Gradle properties are used
+> - 2.1.0: raise this warning to an error
+> - 2.2.0: remove Gradle properties
+
+### Deprecate commonization parameters
+
+> **Issue**: [KT-75161](https://youtrack.jetbrains.com/issue/KT-75161)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The parameters for experimental commonization modes are deprecated in the Kotlin Gradle plugin.
+> These parameters may produce invalid compilation artifacts that are then cached. To delete affected artifacts:
+>
+> 1. Remove the following options from your `gradle.properties` file:
+>
+>    ```none
+>    kotlin.mpp.enableOptimisticNumberCommonization
+>    kotlin.mpp.enablePlatformIntegerCommonization
+>    ```
+>
+> 2. Clear the commonization cache in the `~/.konan/*/klib/commonized` directory or run the following command: 
+>
+>    ```bash
+>    ./gradlew cleanNativeDistributionCommonization
+>    ```
+>
+> **Deprecation cycle**:
+>
+> - 2.2.0: commonization parameters are deprecated with an error
+> - 2.2.20: remove commonization parameters
+
+### Deprecate support for legacy metadata compilation
+
+> **Issue**: [KT-61817](https://youtrack.jetbrains.com/issue/KT-61817)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Options that were used to set up hierarchical structure and create intermediate source sets
+> between the common and intermediate source sets became obsolete.
+> The following compiler options are removed:
+> 
+> * `isCompatibilityMetadataVariantEnabled`
+> * `withGranularMetadata`
+> * `isKotlinGranularMetadataEnabled`
+>
+> **Deprecation cycle**:
+
+> - 2.2.0: remove compiler options from the Kotlin Gradle plugin
+
+### Deprecate `KotlinCompilation.source`
+
+> **Issue**: [KT-64991](https://youtrack.jetbrains.com/issue/KT-64991)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The access to `KotlinCompilation.source` that allowed adding Kotlin source sets directly to the Kotlin
```

Comment:
@danil-pavlov do you mean that the `source()` function is deprecated? It's  not clear what "the access" means for me 🤔

Reply:
This is actually a deprecation cycle update of [this change](https://www.jetbrains.com/help/kotlin-multiplatform-dev/multiplatform-compatibility-guide.html#deprecated-api-for-adding-kotlin-source-sets-directly-to-the-kotlin-compilation), so I tried to be consistent with that text
Let's call it `KotlinCompilation.source` API? to be more general